### PR TITLE
Update gid API return codes.

### PR DESCRIPTION
**Pull request by hidmic**
To be compliant with the API spec. Connected to https://github.com/ros2/rmw/pull/274.

### CODE REVIEW

**Comment by hidmic** [comment]
CI up to `test_rmw_implementation`:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12383/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7352/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10095/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12296/)

File: rmw_fastrtps_shared_cpp/src/rmw_get_gid_for_publisher.cpp [hunk 1]
```diff
@@ -27,28 +28,15 @@ __rmw_get_gid_for_publisher(
...
+  RMW_CHECK_TYPE_IDENTIFIERS_MATCH(
+    publisher,
+    publisher->implementation_identifier,
+    identifier,
+    return RMW_RET_INCORRECT_RMW_IMPLEMENTATION);
+  RMW_CHECK_ARGUMENT_FOR_NULL(gid, RMW_RET_INVALID_ARGUMENT);
 
   auto info = static_cast<const CustomPublisherInfo *>(publisher->data);
-
-  if (!info) {
```
Comment (Lobotuerk) [inline rmw_fastrtps_shared_cpp/src/rmw_get_gid_for_publisher.cpp:47]: You don't need to check if publisher->data was castable or at least not nullptr?

Comment (hidmic) [reply rmw_fastrtps_shared_cpp/src/rmw_get_gid_for_publisher.cpp:47]: Same reply as in https://github.com/ros2/rmw_cyclonedds/pull/244#discussion_r492741744.

**Review by Lobotuerk — APPROVED** [review]
LGTM

**Review by ivanpauno — APPROVED** [review]

**Comment by ivanpauno** [comment]
> CI up to test_rmw_implementation:

Consider running CI up to rcl

**Comment by hidmic** [comment]
CI up to `test_rmw_implementation` and `rcl`:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12401/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7369/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10112/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12314/)

**Comment by hidmic** [comment]
One more time after rebasing [ros2/rmw_cyclonedds#244]( https://github.com/ros2/rmw_cyclonedds/pull/244):

 * Linux [Build Status](http://ci.ros2.org/job/ci_linux/12407/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7374/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10117/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12320/)

**Comment by hidmic** [comment]
Alright, finally green. Going in!

**Pull request merged**